prow/tide/gerrit.go

```diff
@@ -0,0 +1,324 @@
+				crc := CodeReviewCommonFromGerrit(&pr, instance)
+				res[prKey(crc)] = *crc
+			}
+		}
+	}
+
+	return res, nil
+}
+
+func (p *GerritProvider) blockers() (blockers.Blockers, error) {
+	// This is not supported yet, so return an empty blocekr for now.
+	return blockers.Blockers{}, nil
+}
+
+func (p *GerritProvider) isAllowedToMerge(crc *CodeReviewCommon) (string, error) {
+	if crc.Mergeable == string(githubql.MergeableStateConflicting) {
+		return "PR has a merge conflict.", nil
+	}
+	return "", nil
+}
+
+// GetRef gets the latest revision from org/repo/branch.
+func (p *GerritProvider) GetRef(org, repo, ref string) (string, error) {
+	return p.gc.GetBranchRevision(org, repo, ref)
+}
+
+// headContexts gets the status contexts for the commit with OID ==
+// pr.HeadRefOID
+//
+// Assuming all submission requirements are already met as the PRs queried are
+// already submittable. So the focus here is to ensure that all prowjobs were
+// tested against latest baseSHA.
+// Prow parses baseSHA from the `Description` field of a context, will make sure
+// that all Prow jobs that vote to required labels are represented here.
+func (p *GerritProvider) headContexts(crc *CodeReviewCommon) ([]Context, error) {
+	var res []Context
+	pr := crc.Gerrit
+	if pr == nil {
+		return res, errors.New("error not supposed to happen, crc.Gerrit cannot be nil for GerritProvider")
+	}
+
+	selector := map[string]string{
+		kube.GerritRevision:   crc.HeadRefOID,
+		kube.ProwJobTypeLabel: string(prowapi.PresubmitJob),
+		kube.OrgLabel:         crc.Org,
+		kube.RepoLabel:        crc.Repo,
+		kube.PullLabel:        strconv.Itoa(crc.Number),
+	}
+	var pjs v1.ProwJobList
+	if err := p.pjclientset.List(context.Background(), &pjs, ctrlruntimeclient.MatchingLabels(selector)); err != nil {
+		return nil, fmt.Errorf("Cannot list prowjob with selector %v", selector)
+	}
+
+	// keep track of latest prowjobs only
+	latestPjs := make(map[string]*prowapi.ProwJob)
+	for _, pj := range pjs.Items {
+		pj := pj
+		exist, ok := latestPjs[pj.Spec.Context]
+		if ok && exist.CreationTimestamp.After(pj.CreationTimestamp.Time) {
+			continue
+		}
+		latestPjs[pj.Spec.Context] = &pj
+	}
+
+	for _, pj := range latestPjs {
+		res = append(res, Context{
+			Context:     githubql.String(pj.Spec.Context),
+			Description: githubql.String(config.ContextDescriptionWithBaseSha(pj.Status.Description, pj.Spec.Refs.BaseSHA)),
+			State:       githubql.StatusState(pj.Status.State),
+		})
+	}
+
+	return res, nil
+}
+
+func (p *GerritProvider) mergePRs(sp subpool, prs []CodeReviewCommon, dontUpdateStatus *threadSafePRSet) error {
+	p.logger.Info("The merge function hasn't been implemented yet, just logging for now.")
+	return nil
+}
+
+// GetTideContextPolicy gets context policy defined by users + requirements from
+// prow jobs.
+//
+// These are all out of scope for Gerrit, as Gerrit PRs queried are already
+// submittable and mergeable. So return an empty policy option should do.
```

Comment:
This comment is inaccurate and can be removed.

Reply:
they are from previous version of implementation, removed